src/plans/vectorial_plan.jl

```diff
@@ -0,0 +1,286 @@
+@doc raw"""
+    AbstractVectorialType
+
+An abstract type for different representations of a vectorial function
+    ``f: \mathcal → \mathbb R^m`` and its (component-wise) gradient/Jacobian
+"""
+abstract type AbstractVectorialType end
+
+@doc raw"""
+    CoefficientVectorialType{B<:AbstractBasis} <: AbstractVectorialType
+
+# Fields
+
+* `basis` an [`AbstractBasis`](@extref) to indicate the default representation.
+
+"""
+struct CoefficientVectorialType{B<:AbstractBasis} <: AbstractVectorialType
+    basis::B
+end
+@doc raw"""
+    ComponentVectorialType <: AbstractVectorialType
+
+"""
+struct ComponentVectorialType <: AbstractVectorialType end
+
+@doc raw"""
+    FunctionVectorialType <: AbstractVectorialType
+
+"""
+struct FunctionVectorialType <: AbstractVectorialType end
+
+@doc raw"""
+    PowerManifoldVectorialType{<:AbstractPowerRepresentation} <: AbstractVectorialType
+
+Represent that the gradients rare represented as an element of a tangent space from a
+power manifold with a certain [`AbstractPowerRepresentation`](@extref).
+
+# Constructor
+
+    PowerManifoldVectorialType(representation::AbstractPowerRepresentation)
+"""
+struct PowerManifoldVectorialType{<:AbstractPowerRepresentation} <: AbstractVectorialType end
+PowerManifoldVectorialType(::TPR) = PowerManifoldVectorialType{TPR}()
+
+@doc raw"""
+    VectorGradientFunction{E, FT, JT, F, J, I} <: <: AbstractManifoldObjective{E}
+
+Represent a function ``f:\mathcal M → \mathbb R^n`` including its gradient.
```

Comment:
```suggestion
Represent a function ``f:\mathcal M → \mathbb R^n`` including its gradient with respect to each output coordinate.
```
Technically only scalar functions have gradients.